File: Data Structures & Algorithms/binary-search/submission-5.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size();
        while(l < r)
        {
            int mid = (l+r)/2;
            if(nums[mid] == target) 
                return mid;
            if(nums[mid] > target)
                r = mid-1;
            else
                l = mid+1;
        }
        if(l < nums.size() && nums[l] == target) return l;
        if(r >= 0 && nums[r] == target) return r;
        return -1;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-5.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        pfx = [0] * (len(nums) + 1)
        sfx = [0] * (len(nums) + 1)
        pfx[0] = 1
        sfx[len(nums)] = 1
        for i in range(len(nums)):
            pfx[i + 1] = pfx[i] * nums[i]
            sfx[len(nums) - i - 1] = sfx[len(nums) - i] * nums[len(nums) - i - 1]
        res = []
        for i in range(len(nums)):
            res.append(pfx[i] * sfx[i + 1])
        return res